sympy/geometry/point.py

```diff
@@ -398,10 +398,31 @@ def intersection(self, o):
 
         return o.intersection(self)
 
-    def dot(self, p2):
-        """Return dot product of self with another Point."""
-        p2 = Point(p2)
```

Comment:
Then documentation should mention that p2/p is Point or tuple (not just Point) and whether ValueError is thrown otherwise. That is the case for all functions.